src/screens/Asset/ReceiveModal.js

```diff
@@ -138,79 +80,59 @@ const ReceiveModal = ({
+      //     <LabelBadge
+      //       label={t('label.erc20TokensOnly')}
+      //       labelStyle={{ fontSize: fontSizes.tiny }}
+      //       primary
+      //     />
+      //   ),
+      // }] : undefined}
       centerFloatingItem={
         <ImageWrapper style={{ position: 'absolute', marginTop: -24 }}>
-          <ProfileImage
-            uri={profileImageURI}
-            userName={username}
-            diameter={48}
-          />
+          <ProfileImage uri={profileImageURI} userName={username} diameter={48} />
         </ImageWrapper>
       }
     >
       <ContentWrapper forceInset={{ top: 'never', bottom: 'always' }}>
-        <WarningBanner rounded small />
-        {!!ensName && (
-          <InfoView>
-            <BaseText
-              big
-              onPress={() => handleCopyToClipboard(ensName, true)}
-              center
-            >
+        <InfoView>
+          {!!ensName && (
+            <TextWithCopy textToCopy={ensName} ensCopy iconColor={baseColors.dodgerBlue}>
               {ensName}
-            </BaseText>
-            <BaseText regular center secondary>{t('label.yourEnsName')}</BaseText>
-          </InfoView>
-        )}
-        <QRCodeWrapper>
-          <View style={{ overflow: 'hidden', padding: 10 }}>
-            {!!address && <QRCodeWithTheme value={address} size={160} />}
-          </View>
-          <WalletAddress onPress={() => handleCopyToClipboard(address)}>
-            {address}
-          </WalletAddress>
-        </QRCodeWrapper>
-        <ButtonsRow>
-          {showBuyTokensButton && (
-          <Button
-            title={t('button.buyTokens')}
-            onPress={handleBuyTokens}
-            primarySecond
-            small={needsSmallButtons}
-            style={{ flex: 1, marginRight: 10 }}
-          />
+            </TextWithCopy>
           )}
-          <Button
-            title={t('button.shareAddress')}
-            onPress={handleAddressShare}
-            small={needsSmallButtons}
-            style={{ flex: 1 }}
-          />
-        </ButtonsRow>
-        {showBuyTokensButton && (
-        <IconsContainer>
-          <Image source={visaIcon} />
-          <IconsSpacing />
-          <Image source={mastercardIcon} />
-        </IconsContainer>
-        )}
+          {ensName ? (
+            <WalletAddress>{address}</WalletAddress>
+          ) : (
+            <WalletAddressWithCopy textToCopy={address} iconColor={baseColors.dodgerBlue}>
+              {address}
+            </WalletAddressWithCopy>
+          )}
+        </InfoView>
+        <QRCodeWrapper>{!!address && <QRCodeWithTheme value={address} size={104} />}</QRCodeWrapper>
+        <WarningText center small>
+          {t('paragraph.cautionMessage', {
+            ethereum: t('chains.ethereum'),
+            mediumText: true,
+            color: baseColors.desaturatedDarkBlue,
+          })}{' '}
+          <BaseText style={{ color: baseColors.dodgerBlue }}>{t('paragraph.withCaution')}</BaseText>
```

Comment:
Not sure if with caution was intended to be a link. @IAmKio wdyt?

Reply:
@mdjastrzebski cc: @divyaBacancy 
That was definitely intended to be a link to the Interjection UI.

Reply:
So the warning should be displayed only when ethereum wallet is not deployed.

Reply:
@IAmKio: receive modal opens interjection ui which offers to open receive modal which opens interjection ui (-:

Reply:
Done.